bokehjs/src/lib/models/tools/math_text.ts

```diff
@@ -0,0 +1,130 @@
+import * as p from "core/properties"
+import {isNumber} from "core/util/types"
+import {Context2d} from "core/util/canvas"
+import {ImageLoader} from "core/util/image"
+import {CanvasImage} from "models/glyphs/image_url"
+import {Model} from "../../model"
+
+
+declare namespace MathJax {
+  function tex2svg(input: string): HTMLElement
+}
+
+export namespace MathText {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = Model.Props & {
+    text: p.Property<string>
+  }
+}
+
+export interface MathText extends MathText.Attrs {}
+
+type Position = {
+  sx: number
+  sy: number
+  x_anchor?: number | "left" | "center" | "right"
+  y_anchor?: number | "top"  | "center" | "baseline" | "bottom"
+}
+
+export class MathText extends Model {
```

Comment:
This model mixes model and view code. Those need to be separated and a view created. Then you will have to build and use that view along its model in axis view. In general, models must not include code that's dependent on the main thread browser environment, as models are intended to be used in nodejs and workers. There are models that are implemented against this advice (e.g. remote sources), but those will be converted at some point.

Reply:
Separating the view and the model is no problem, I got that. But I use the model as the axis_label property of Axis, how would I use a view there? 
Any examples?

Reply:
You need to `build_view()` the model in `AxisView.lazy_initialize()` and then use the constructured view. Grep the repository for `build_view()` (or `build_views()` for more context).